src/main/resources/static/assets/js/scripts.js

```diff
@@ -48,32 +48,78 @@ function confirm_delete_category(id) {
     }
 }
 
-<!-- Validate Save Category-->
+<!-- Validate Add Category-->
 $(document).ready(function() {
     $("#newCategory #categoryName").keyup(function() {
-        document.getElementById("errorCategory").style.display = "none";
-        document.getElementById("saveCategory").disabled = false;
-        var name = $("#newCategory #categoryName").val();
-        if (name == '') {
-            errorSaveCategory("Required field! Please type something!");
-            return;
-        };
-        if (name.length < 3 || name.length > 20) {
-            errorSaveCategory("The category name must be 3 to 20 characters in length!");
-            return;
-        };
-        axios.get('/categories/' + name)
+        document.getElementById("errorAddCategory").style.display = "none";
+        document.getElementById("submitAddCategory").disabled = false;
+        var inputName = $("#newCategory #categoryName").val();
+        var message = validateName(inputName);
+        if (message != null) {
+            return errorEditCategory(message);
+        }
```

Comment:
if (!validateName(inputName) {
  return error...;
}
and validate name should return 1 or 0. don't you think it's cleaner this way?